waltz-test/src/main/python/waltz_ducktape/tests/validation/connection_interruption.py

```diff
@@ -0,0 +1,38 @@
+from threading import Thread
+from time import sleep
+
+
+class ConnectionInterruption(Thread):
+    """
+    A scheduler to drop incoming traffic to a port periodically.
+    """
+
+    def __init__(self, service, interruption_length, num_of_interruptions, service_node_index):
+        """
+            Construct a new 'ConnectionInterruption' object.
+
+            :param service: The service having incoming traffic to a port blocked
+            :param interruption_length: The interval(milliseconds) between two interruptions and the length of each interruption
+            :param num_of_interruptions: Total number of network connection interruptions
+            :param service_node_index: A service node used for this test
+        """
+        Thread.__init__(self)
+        self.service = service
+        self.interruption_length = interruption_length
+        self.num_of_interruptions = num_of_interruptions
+        self.service_node_index = service_node_index
+
+    def run(self):
+        node = self.service.nodes[self.service_node_index]
+        for interruption in range(self.num_of_interruptions):
+            # let waltz server receive transactions as usual
+            sleep(self.interruption_length)
```

Comment:
What does "interruption_length" refer to? Is it the amount of time that the network connection will be down?
If yes, then what is the reason behind this statement? 
May be we can add a separate parameter to provide the frequency of the interruption and use that variable here?